feedhenry/fh-ios-sdk
# fh-ios-sdk project should contain only library+tests target

Motivation:
Currently the fh-ios-sdk xcode project apart from the library contained demo targets.  This PR extracts the library as a standalone target including the tests. The demo included in the project was extracted into a separate demo project. All other demos in the repo were moved in the /demos folder. Eventually these demos will be moved out from the repo too.

To test:
Simple open ```fh-ios-sdk.xcodeproj``` and run Product->Tests. Project should be build and tests run successfully. Further the ```build.sh``` script was adjusted and should create the framework build plus the starter project (as before) in the ```/dist``` folder.

Note the /dist folder was removed from the repo. As discussed with @wei-lee  we can use the 'Release' feature of Github to publish the released artifacts instead of holding them in the repo itself.

@wei-lee mind to review?
